bin/adhoc-scripts/updatePileupObjects.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+This script can be used to fetch and update pileup objects through the
+MSPileup REST API.
+
+The user provides the MSPileup URL and a JSON file as input, which will be
+used to update every single pileup document that it fetches from MSPileup.
+
+Example usage is:
+python3 updatePileupObjects.py --url=https://cmsweb.cern.ch --fin=override_data.json
+
+NOTE: due to Python3 urllib/ssl implementation in order to access data from
+CMSWEB we must provide valid cert/key pem files AND CERN CA file which usually
+located at:
+    /etc/grid-security/certificates/CERN-Root-2.pem
+Without CERN CA the user's certificate/proxy are considered as self-signed and
+ssl fails to proceed and verify client's certificate.
+"""
+import argparse
+import json
+import logging
+import os
+import sys
+import urllib.request
+import ssl
+from pprint import pprint
+
+
+class OptionParser():
+    """Class to parse the command line arguments"""
+
+    def __init__(self):
+        "User based option parser"
+        self.parser = argparse.ArgumentParser(prog='PROG')
+        self.parser.add_argument("--fin", required=True,
+                                 dest="fin", help="Input JSON file")
+        self.parser.add_argument("--dry-run", action="store_true",
+                                 dest="dryrun", help="Fetch docs but do not update")
+        self.parser.add_argument("--url", action="store",
+                                 dest="url", default="https://cmsweb-testbed.cern.ch",
+                                 help="URL for the MSPileup service")
+        self.parser.add_argument("--cafile", required=False,
+                                 dest="cafile", help="CERN CA file")
+
+
+def getContext(client_cert, client_key, cafile):
+    """
+    Creates and return SSL context used by urllib module
+
+    :param client_cert: client's certificate pem file or proxy
+    :param client_key: client's key pem file or proxy
+    :param cafile: CERN CA certificate
+    :return: SSL context used by urllib
+    """
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+    context.load_cert_chain(certfile=client_cert, keyfile=client_key)
+    if cafile:
```

Comment:
Thanks for example. Turns out, urllib library has different implementation. The one you pointed relies on [OpenerDirector](https://docs.python.org/3/library/urllib.request.html#urllib.request.OpenerDirector) instance which can be initialized via external handler. And, its implementation most likely loose in terms of strict standard of validating client/server via HTTPs handshake. While urllib request one is more strict and requires check on client's certificate CA. Said that, I refactor the code to use opener since it does not require CERN Root CA file (which I agree is not convenient).